scripts/sonar/sonar-controller.js

```diff
@@ -41,23 +50,51 @@ angular.module('app.sonar').controller('SonarCtrl', ['$scope', '$log', '$q', 'ap
                 var totalLines = sum(timePeriod.metrics, 'lines');
 
                 var weightedMetrics = timePeriod.metrics.map(function(project) {
-                    var factor = project.lines / totalLines;
 
                     return {
-                        methodComplexity: factor * project.methodComplexity,
-                        fileComplexity: factor * project.fileComplexity,
-                        coverage: factor * project.coverage,
-                        compliance: factor * project.compliance,
+                        numberOfMethodsInProjectInPeriod: project.totalMethods,
+                        numberOfFilesInProjectInPeriod: project.totalFiles,
+                        complexityInProjectInPeriod: project.totalComplexity,
+                        testableLinesInProjectInPeriod: project.testableLines,
+                        untestedLinesInProjectInPeriod: project.untestedLines,
+
+                        weightedBlockingIssuesInProjectInPeriod: weighting.blocker * project.blockerIssues,
+                        weightedCriticalIssuesInProjectInPeriod: weighting.critical * project.criticalIssues,
+                        weightedMajorIssuesInProjectInPeriod: weighting.major * project.majorIssues,
+                        weightedMinorIssuesInProjectInPeriod: weighting.minor * project.minorIssues,
+                        weightedInfoIssuesInProjectInPeriod: weighting.info * project.infoIssues
+
                     };
                 });
-
+                
+                // Complexity calculations
+                var complexityInPeriod = sum(weightedMetrics, 'complexityInProjectInPeriod')
+                
+                var methodComplexityInPeriod = complexityInPeriod / sum(weightedMetrics, 'numberOfMethodsInProjectInPeriod')
+                var fileComplexityInPeriod = complexityInPeriod / sum(weightedMetrics, 'numberOfFilesInProjectInPeriod')
+                
+                // Compliance calculations
+                var totalWeightedIssuesInPeriod = sum(weightedMetrics, 'weightedBlockingIssuesInProjectInPeriod') + sum(weightedMetrics, 'weightedCriticalIssuesInProjectInPeriod')
+                                                + sum(weightedMetrics, 'weightedMajorIssuesInProjectInPeriod') + sum(weightedMetrics, 'weightedMinorIssuesInProjectInPeriod')
+                                                + sum(weightedMetrics, 'weightedInfoIssuesInProjectInPeriod');
+
+                var rulesComplianceInPeriod = (100 - (totalWeightedIssuesInPeriod / totalLines) * 100)
+                
+                // Coverage calculations
+                var testableLinesOfCodeInPeriod = sum(weightedMetrics, 'testableLinesInProjectInPeriod')
+                var untestedLinesOfCodeInPeriod = sum(weightedMetrics, 'untestedLinesInProjectInPeriod')
+                
+                var codeCoverageInPeriod = ((testableLinesOfCodeInPeriod - untestedLinesOfCodeInPeriod) / testableLinesOfCodeInPeriod) * 100
```

Comment:
Yep, my comment above there is off, that gives you the percentage of code uncovered, not covered. That should be subtracted from 100 to give a coverage percentage.

It could also be written as:

    var codeCoverage = ((testableLines - uncoveredLines) / testableLines) * 100 || 0;